packages/modules/e3dc/device.py

```diff
@@ -0,0 +1,139 @@
+                         id=num)
+    log.debug("Konfig: %s", device_config)
+    return device_config
+
+
+def read_legacy_counter(address1: str,
+                        address2: str,
+                        read_ext: int,
+                        pvinv: str,
+                        num: int):
+    component_config = E3dcCounterSetup(configuration=E3dcCounterConfiguration())
+    component_config.id = num
+    run_device_legacy(create_legacy_device_config(address1,
+                                                  read_ext,
+                                                  num), component_config)
+
+
+def read_legacy_bat(address1: str,
+                    address2: str, read_ext: int,
+                    pvinv: str,
+                    num: int) -> None:
+    # für openwbv19 können mit der bisherigen parametrisierung zwei ip_addressen
+    # ausgelesen werden
+    # ebenso wird bei Speicheraufruf beides (Speicher und PV ausgelesen
+    # in openwb v2.0 geht nur noch eine IP adresse und die Pv muss
+    # separate ausgelesen werden
+    addresses = [address for address in [address1, address2] if address != "none"]
+    log.debug('e3dc IP-Adresse1: ' + address1)
+    log.debug('e3dc IP-Adresse2: ' + address2)
+    log.debug('e3dc read_ext: ' + str(read_ext))
+    log.debug('e3dc pvinv: ' + pvinv)
+    log.debug('e3dc id: ' + str(num))
+    soc = 0
+    power = 0
+    pv_external = 0
+    pv = 0
+    pv_other = pvinv != "none"
+    for address in addresses:
+        log.debug("Ip: %s, read_external %d pv_other %s", address, read_ext, pv_other)
+        with modbus.ModbusTcpClient_(address, port=502) as client:
+            soc_tmp, power_tmp = read_bat(client)
+            soc += soc_tmp
+            power += power_tmp
+            pv_tmp, pv_external_tmp = read_inverter(client, read_ext)
+            pv += pv_tmp
+            pv_external += pv_external_tmp
+    soc /= len(addresses)
+    log.debug("Soc %d power %d pv %d pv_external %d",
+              soc, power, pv, pv_external)
+    counter_import, counter_export = sim_count(power, prefix="speicher")
+    get_bat_value_store(1).set(BatState(power=power, soc=soc, imported=counter_import, exported=counter_export))
+    # pv_other sagt aus, ob WR definiert ist, und dessen PV Leistung auch gilt
+    # wenn 0 gilt nur PV und pv_external aus e3dc
+    pv_total = pv + pv_external
+    # Wenn wr1 nicht definiert ist, gilt nur die PV Leistung die hier im Modul ermittelt wurde
+    # als gesamte PV Leistung für wr1
+    # Wenn wr1 definiert ist, gilt die bestehende PV Leistung aus Wr1 und das was hier im Modul ermittelt wurde
+    # als gesamte PV Leistung für wr1
+    if pv_other:
+        pv_total = pv_total + files.pv[0].power.read()
+    log.debug("wr update pv_other %s pv_total %d", pv_other, pv_total)
+    _, exported_pv = sim_count(pv_total, prefix="pv")
+    get_inverter_value_store(1).set(InverterState(exported=exported_pv, power=pv_total))
```

Comment:
```suggestion
    get_inverter_value_store(num).set(InverterState(exported=exported_pv, power=pv_total))
```
`num` ist zwar aktuell sowieso immer 1, aber vielleicht will ja mal jemand die Möglichkeit für den 2er-Slot einbauen. Wenn du `num` nicht verwendest und das immer 1 ist, dann könnte alternativ der ganze Parameter weg.

Reply:
geändert